### PR TITLE
IIS Rewrite (Rebase)

### PR DESCRIPTION
**Pull request by petems**
https://github.com/voxpupuli/puppet-iis/pull/87 (Rebased against master)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,8 @@
 ---
 sudo: false
 language: ruby
-cache: bundler
 bundler_args: --without system_tests
-before_install: rm Gemfile.lock || true
-script:
-  - 'bundle exec rake $CHECK'
+script: "bundle exec rake validate && bundle exec rake lint && bundle exec rake spec SPEC_OPTS='--color --format documentation'"
```
Comment (bastelfreak) [inline .travis.yml]: this will overwrite our modulesync default?

File: README.md [hunk 2]
```diff
@@ -1,161 +1,171 @@
...
+Sending a refresh event to an iis_site type will recycle the web site.
+
+### iis_pool
+
+Enumerate all IIS application pools:
+* `puppet resource iis_pool`
+
+Example output for `puppet resource iis_site 'DefaultAppPool'`
+```
+iis_pool { 'DefaultAppPool':
```
Comment (bastelfreak) [inline README.md]: we could add ```puppet here to force github to do the correct syntax highlighting. but that is probably not needed.

File: .travis.yml (hunk 1 above)
Comment (petems) [inline .travis.yml]: Probably, I can remove this I think

Comment (petems) [inline .travis.yml]: Fixed

File: README.md (hunk 2 above)
Comment (petems) [inline README.md]: Good idea: fixed

**Comment by igalic** [comment]
so, essentially, this is now ready to be merged.
i suggest we release a stable version of the module, as-is, and create a branch for maintenance.
this, we'll merge into master, and then see from there.

**Comment by petems** [comment]
@igalic Sounds good to me. I think we should merge #96 first, so we have a base set of acceptance testing that works in a known good state. If there's any regressions that come up after the rework and 2.X release, we can add in acceptance tests to fix them in the future :+1:

**Comment by jyaworski** [comment]
#96 is merged.

**Comment by petems** [comment]
@jyaworski Yep, we're waiting on author to respond on #89.

Idea: wait until end of the week for a response, if none, release the last 2.X branch, merge this, release the 3.X type/provider version and try to refactor the recycle numbers to work with the new type and provider code.

**Comment by cyberious** [comment]
LGTM

File: .fixtures.yml [hunk 3]
```diff
@@ -1,5 +1,6 @@
 fixtures:
   repositories:
     stdlib: https://github.com/puppetlabs/puppetlabs-stdlib
+    dism: https://github.com/puppetlabs/puppetlabs-dism
```
Comment (cyberious) [inline .fixtures.yml]: Actually not required, it was a idea about downloading and installing via DISM since it has been reborn for Microsoft.

Comment (petems) [inline .fixtures.yml]: Fixed

**Comment by petems** [comment]
YEAH BOIIIIIII

**Pull request merged**